<?xml version="1.0" encoding="UTF-8" ?>
<launch>

    <arg name="bag" default="$(dirname)/../data/bags/traversability/marv/ugv_2022-08-12-15-30-22.bag"/>

    <arg name="params" default="$(eval bag.split()[0] + '.params')"/>
    <arg name="device" default="cuda" doc="Device to run tensor operations on: cpu or cuda."/>

    <param name="use_sim_time" value="true"/>

    <rosparam command="load" file="$(arg params)"/>
    <node name="rosbag_play" pkg="rosbag" type="play" args="--clock --delay 3.0 --rate 1.0 --start 0 $(arg bag)"/>

    <include file="$(find cras_ouster_driver)/launch/os_cloud_node.launch"/>

    <node name="points_destagger" pkg="nodelet" type="nodelet"
          args="standalone cras_ouster_driver/destagger_nodelet">
        <remap from="info" to="/os_node/sensor_info"/>
        <remap from="input" to="/os_cloud_node/points"/>
        <remap from="output" to="/os_cloud_node/destaggered_points"/>
    </node>

    <node name="save_clouds" pkg="traversability_estimation" type="save_clouds" output="screen"/>

</launch>
